tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/BackendTestSettings.scala

```diff
@@ -0,0 +1,234 @@
+package org.apache.spark.sql.rapids.utils
+
+import java.util
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+import com.nvidia.spark.rapids.TestStats
+
+import org.apache.spark.sql.RapidsTestConstants.RAPIDS_TEST
+
+abstract class BackendTestSettings {
+
+  private val enabledSuites: java.util.Map[String, SuiteSettings] = new util.HashMap()
+
+  protected def enableSuite[T: ClassTag]: SuiteSettings = {
+    val suiteName = implicitly[ClassTag[T]].runtimeClass.getCanonicalName
+    if (enabledSuites.containsKey(suiteName)) {
+      throw new IllegalArgumentException("Duplicated suite name: " + suiteName)
+    }
+    val suiteSettings = new SuiteSettings
+    enabledSuites.put(suiteName, suiteSettings)
+    suiteSettings
+  }
+
+  private[utils] def shouldRun(suiteName: String, testName: String): Boolean = {
+    if (!enabledSuites.containsKey(suiteName)) {
+      return false
+    }
+
+    val suiteSettings = enabledSuites.get(suiteName)
+
+    suiteSettings.disableReason match {
+      case Some(_) => return false
+      case _ => // continue
+    }
+
+    val inclusion = suiteSettings.inclusion.asScala
+    val exclusion = suiteSettings.exclusion.asScala
+
+    if (inclusion.isEmpty && exclusion.isEmpty) {
+      // default to run all cases under this suite
+      return true
+    }
+
+    if (inclusion.nonEmpty && exclusion.nonEmpty) {
+      // error
+      throw new IllegalStateException(
+        s"Do not use include and exclude conditions on the same test case: $suiteName:$testName")
+    }
+
+    if (inclusion.nonEmpty) {
+      // include mode
+      val isIncluded = inclusion.exists(_.isIncluded(testName))
+      return isIncluded
+    }
+
+    if (exclusion.nonEmpty) {
+      // exclude mode
+      val isExcluded = exclusion.exists(_.isExcluded(testName))
+      return !isExcluded
+    }
+
+    throw new IllegalStateException("Unreachable code")
+  }
+
+  final protected class SuiteSettings {
+    private[utils] val inclusion: util.List[IncludeBase] = new util.ArrayList()
+    private[utils] val exclusion: util.List[ExcludeBase] = new util.ArrayList()
+
+    private[utils] var disableReason: Option[String] = None
+
+    def include(testNames: String*): SuiteSettings = {
+      inclusion.add(Include(testNames: _*))
+      this
+    }
+    def exclude(testNames: String*): SuiteSettings = {
+      exclusion.add(Exclude(testNames: _*))
+      this
+    }
+    def includeRapidsTest(testName: String*): SuiteSettings = {
+      inclusion.add(IncludeRapidsTest(testName: _*))
+      this
+    }
+    def excludeRapidsTest(testName: String*): SuiteSettings = {
+      exclusion.add(ExcludeRapidsTest(testName: _*))
+      this
+    }
+    def includeByPrefix(prefixes: String*): SuiteSettings = {
+      inclusion.add(IncludeByPrefix(prefixes: _*))
+      this
+    }
+    def excludeByPrefix(prefixes: String*): SuiteSettings = {
+      exclusion.add(ExcludeByPrefix(prefixes: _*))
+      this
+    }
```

Comment:
Would be nice if these took a "reason" string argument, similar to how we set this when skipping/xfailing in our integration tests.  That way the code encourages self-documenting why tests are being excluded which should not be the norm.  Can also emit this reason when logging tests that are skipped/excluded.